src/postgres.rs

```diff
@@ -117,10 +118,35 @@ impl PostgresSession {
         }
         const NUMBER_OF_ARGS: usize = 10;
 
-        let mut args: Vec<&(dyn ToSql + Sync)> = Vec::with_capacity(NUMBER_OF_ARGS * txs.len());
         let txs: Vec<PostgresTransactionInfo> =
             txs.iter().map(PostgresTransactionInfo::from).collect();
+
+        let statement = format!(
+            r#"
+                COPY banking_stage_results.transaction_infos(
+                    signature, errors, is_executed, is_confirmed, first_notification_slot, cu_requested, prioritization_fees, utc_timestamp, accounts_used, processed_slot
+                ) FROM STDIN BINARY
+            "#,
+        );
+        let sink: CopyInSink<bytes::Bytes> = self.copy_in(&statement).await.unwrap();
+        let writer = BinaryCopyInWriter::new(
+            sink,
+            &[
+                Type::TEXT,
+                Type::CHAR,
+                Type::CHAR,
+                Type::INT8,
+                Type::INT8,
+                Type::INT8,
+                Type::TIMESTAMPTZ,
+                Type::TEXT,
+                Type::INT8,
+                Type::TEXT,
+            ],
+        );
+        pin_mut!(writer);
         for tx in txs.iter() {
+            let mut args: Vec<&(dyn ToSql + Sync)> = Vec::with_capacity(NUMBER_OF_ARGS);
```

Comment:
I would create the slice right away without Vec/push

Reply:
The code exisits lazy to remove the vector and sloce